lib/ohai/plugins/vmware.rb

```diff
@@ -40,28 +41,30 @@ def from_cmd(cmd)
   end
 
   def get_vm_attributes(vmtools_path)
-    if !File.exist?(vmtools_path)
-      Ohai::Log.debug("#{vmtools_path} not found")
-    else
-      vmware Mash.new
-      begin
-        # vmware-toolbox-cmd stat <param> commands
-        # Iterate through each parameter supported by the "vwware-toolbox-cmd stat" command, assign value
-        # to attribute "vmware[:<parameter>]"
-        [ "hosttime", "speed", "sessionid", "balloon", "swap", "memlimit", "memres", "cpures", "cpulimit" ].each do |param|
-          vmware[param] = from_cmd("#{vmtools_path} stat #{param}")
-          if vmware[param] =~ /UpdateInfo failed/
-            vmware[param] = nil
+    if virtualization[:systems][:vmware]
```

Comment:
You can avoid indenting the entirety of this method by pairing this check with the call to the method:

```ruby
collect_data(:linux) do
  get_vm_attributes("/usr/bin/vmware-toolbox-cmd") if virtualization[:systems][:vmware]
end
```